ospf6d/ospf6d.c

```diff
@@ -1161,54 +1293,141 @@ DEFUN (show_ipv6_ospf6_border_routers,
+      "Specify Router ID as IPv4 address notation\n"
+      "Display Network Entry\n"
+      "Specify Router ID as IPv4 address notation\n"
+      "Specify Link state ID as IPv4 address notation\n")
 {
 	int idx_ipv4 = 5;
-	struct listnode *node;
+	struct listnode *node, *nnode;
 	struct ospf6_area *oa;
 	struct ospf6 *ospf6 = NULL;
+	const char *vrf_name = VRF_DEFAULT_NAME;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+
+	OSPF6_CMD_CHECK_RUNNING();
+	OSPF6_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+	if (idx_vrf > 0)
+		idx_ipv4 += 2;
+
+	if (all_vrf) {
+		for (ALL_LIST_ELEMENTS_RO(om6->ospf6, nnode, ospf6)) {
+			for (ALL_LIST_ELEMENTS_RO(ospf6->area_list, node, oa)) {
+				vty_out(vty,
+					"\n        SPF Result in Area %s\n\n",
+					oa->name);
+				ospf6_linkstate_table_show(vty, idx_ipv4, argc,
+							   argv, oa->spf_table);
+			}
+			vty_out(vty, "\n");
+		}
+		return CMD_SUCCESS;
+	}
+	ospf6 = ospf6_lookup_by_vrf_name(vrf_name);
+	if (ospf6 == NULL) {
+    vty_out(vty, "%% OSPF6 instance not found\n");
+    return CMD_SUCCESS;
+	}
 
-	ospf6 = ospf6_lookup_by_vrf_name(VRF_DEFAULT_NAME);
-	OSPF6_CMD_CHECK_RUNNING(ospf6);
 	for (ALL_LIST_ELEMENTS_RO(ospf6->area_list, node, oa)) {
 		vty_out(vty, "\n        SPF Result in Area %s\n\n", oa->name);
 		ospf6_linkstate_table_show(vty, idx_ipv4, argc, argv,
 					   oa->spf_table);
 	}
-
 	vty_out(vty, "\n");
+
 	return CMD_SUCCESS;
 }
 
 
-DEFUN (show_ipv6_ospf6_linkstate_detail,
-       show_ipv6_ospf6_linkstate_detail_cmd,
-       "show ipv6 ospf6 linkstate detail",
-       SHOW_STR
-       IP6_STR
-       OSPF6_STR
-       "Display linkstate routing table\n"
-       "Display detailed information\n")
+DEFUN(show_ipv6_ospf6_linkstate_detail, show_ipv6_ospf6_linkstate_detail_cmd,
+      "show ipv6 ospf6 [vrf <NAME|all>] linkstate detail",
+      SHOW_STR IP6_STR OSPF6_STR VRF_CMD_HELP_STR
+      "All VRFs\n"
+      "Display linkstate routing table\n"
+      "Display detailed information\n")
 {
 	int idx_detail = 4;
 	struct listnode *node;
 	struct ospf6_area *oa;
 	struct ospf6 *ospf6 = NULL;
-
-	ospf6 = ospf6_lookup_by_vrf_name(VRF_DEFAULT_NAME);
-	OSPF6_CMD_CHECK_RUNNING(ospf6);
+	const char *vrf_name = VRF_DEFAULT_NAME;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+
+
+	OSPF6_CMD_CHECK_RUNNING();
+	OSPF6_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+	if (idx_vrf > 0)
+		idx_detail += 2;
+	if (all_vrf) {
+		for (ALL_LIST_ELEMENTS_RO(om6->ospf6, node, ospf6)) {
+			for (ALL_LIST_ELEMENTS_RO(ospf6->area_list, node, oa)) {
+				vty_out(vty,
+					"\n        SPF Result in Area %s\n\n",
+					oa->name);
+				ospf6_linkstate_table_show(vty, idx_detail,
+							   argc, argv,
+							   oa->spf_table);
+			}
+			vty_out(vty, "\n");
+		}
+		return CMD_SUCCESS;
+	}
+	ospf6 = ospf6_lookup_by_vrf_name(vrf_name);
+	if (ospf6 == NULL) {
+    vty_out(vty, "%% OSPF6 instance not found\n");
+    return CMD_SUCCESS;
+	}
 
 	for (ALL_LIST_ELEMENTS_RO(ospf6->area_list, node, oa)) {
```

Comment:
Same comment as before. This call for specific VRFs could also have been accommodated inside the for loop for 'all_vrf'.

Reply:
Resolved